recOrder/scripts/cli.py

```diff
@@ -1,148 +1,181 @@
-        )
+        def text_overlay():
+            v.text_overlay.text = (
+                f"Channel: {slice_names[v.dims.current_step[1]]}"
+            )
 
-        ptzyx = (len(position),) + (reader.shape[0],) + reader.shape[2:]
-        for j in range(int(reader.channels)):
-            temp_data = np.zeros(ptzyx)
-            for k, pos in enumerate(position):
-                temp_data[k] = reader.get_array(pos)[:, j, ...]
-            v.add_image(temp_data, name=reader.channel_names[j])
-            v.layers[-1].reset_contrast_limits()
-        v.dims.axis_labels = ("P", "T", "Z", "Y", "X")
+    v.dims.events.current_step.connect(text_overlay)
+    text_overlay()
 
     napari.run()
 
 
+@click.group()
+def cli():
+    print(
+        "\033[92mrecOrder: Computational Toolkit for Label-Free Imaging\033[0m\n"
+    )
+
+
 @cli.command()
 @click.help_option("-h", "--help")
+@click.argument("filename")
 @click.option(
-    "--input",
-    required=True,
-    type=click.Path(exists=True),
-    help="path to the raw data folder containing ome.tifs",
-)
-@click.option(
-    "--output",
-    required=True,
-    type=str,
-    help="full path to save the zarr store (../../Experiment.zarr)",
-)
-@click.option(
-    "--data_type",
-    required=False,
-    type=str,
-    help='Data type, "ometiff", "upti", "zarr"',
-)
-@click.option(
-    "--replace_pos_name",
-    required=False,
-    type=bool,
-    help="whether or not to append position name to data",
+    "--positions",
+    "-p",
+    default=None,
+    multiple=True,
+    help="""For ome-zarr v0.4 datasets, position name string e.g. \"-p 0/0/0\"
+for the 0th row, 0th column, and 0th fov. For other datasets, integer positions
+e.g. \"-p 0\" for the 0th position. Accepts multiple positions e.g. \"-p 0/0/0
+-p 0/1/10\".""",
 )
 @click.option(
-    "--format_hcs",
-    required=False,
-    type=bool,
-    help='whether or not to format the data as an HCS "well-plate"',
+    "--layers",
+    "-l",
+    default="position",
+    type=click.Choice(["position", "channel", "p", "c"]),
+    help="Layers as 'position' ('p') or 'channel' ('c')",
 )
-def convert(input, output, data_type, replace_pos_name, format_hcs):
-    """Convert MicroManager ome-tiff to ome-zarr"""
-    from recOrder.io.zarr_converter import ZarrConverter
+def view(filename, positions=None, layers=None):
+    """View a dataset in napari"""
+    click.echo(f"Reading file:\t {filename}")
+    print_info(filename)
 
-    converter = ZarrConverter(
-        input, output, data_type, replace_pos_name, format_hcs
-    )
-    converter.run_conversion()
+    reader = _get_reader(filename)
+
+    if positions == ():  # If no positions specified, open all positions
+        positions = _build_complete_position_list(reader)
+
+    arrays = _build_array_list(reader, positions, layers)
+    layer_names, slice_names = _build_name_lists(reader, positions, layers)
+
+    _create_napari_viewer(arrays, layers, layer_names, slice_names)
```

Comment:
@ziw-liu I'd appreciate any general comments or advice on this refactor. Parts of it are inspired by your `napari-ome-zarr` reader, but it's mostly from scratch.

You mentioned that you're handling `Wells` as well? I'm not sure I fully understood that.

Reply:
> You mentioned that you're handling `Wells` ~as well~ too? I'm not sure I fully understood that.

I gathered all the FOVs in a well and arranged them in a multi-scale list of (stack or grid of dask arrays) as layer data.

Reply:
I'm having trouble handling both `Plates` and `Positions` smoothly...see all of the `if/else` statements I had to introduce here:

[a0dedde](https://github.com/mehta-lab/recOrder/pull/331/commits/a0dedde2f7641ebc2cd168c662a606a0a064726e)

Is there a "unified" way to access all positions for a `Plate` object or a `Position` object (which has only a single position)? Can `positions()` become a method of `NGFFNode` so that I can use it for both `Plate` and `Position` objects?

Reply:
I think for a viewer that supports both one image and multiple nested images, this logic is somewhat inevitable -- the data structures are actually different. The ome-zarr-py/napari-ome-zarr implementation alleviates some of it by breaking specification compliance and impose extra restrictions on how data is stored. Since recOrder does not necessarily need to be a general purpose viewer for NGFF, imposing some restrictions may not be unacceptable.

> Can `positions()` become a method of `NGFFNode` so that I can use it for both `Plate` and `Position` objects?

Further discussion can be an iohub issue. But in short, it will be very confusing for me to see that `Position` (which represents one position) has a `positions()` method which implies multiple positions.

Reply:
Great, thanks.